src/edu/csus/ecs/pc2/core/execute/Executable.java

```diff
@@ -3382,4 +3519,87 @@ private String formatTestCasePhase(String runPhase, int testCase)
     {
         return(String.format("%10s test case %s", runPhase, StringUtilities.rpad(' ', 5, Integer.toString(testCase) + "...")));
     }
+    
+    /**
+     * Get the filename of a judge's file.  This could be a full path if the data files are external.  If the
+     * files are internal, then we return the first one's name since each dataset will get copied to the name
+     * of the first item.
+     * 
+     * @return filename of a judge's file
+     * @param wantInput - boolean indicating if we want the input file (true).  If false, we want the answer file
+     * @param setIndex - Which dataset element number are we interested in [0, #datasets-1]
+     */
+    private String getJudgeFileName(Utilities.DataFileType type, int setIndex)
```

Comment:
FIX: move this logic into JudgeUtilties so only a single line add/change in Executable.

Reply:
The problem is, `getJudgeFileName()` requires several context variables from `Executable`: `problem`, `problemDataFiles` and `log`  - these would have to be passed into the utility class method,  making it a very specific utility class method useful only to a caller that has "all those things", eg. `Executable`.  Do we want to do that?  That is, do we want:

    public static String getJudgeFileName(Utilities.DataFileType type, int setIndex, Problem problem, ProblemDataFiles problemDataFiles, Log log)

as the utility method?

Reply:
I will add this to my refactor list, no change needed at this time.